core/shared/src/main/scala/zio/FiberRefs.scala

```diff
@@ -26,50 +27,78 @@ import scala.annotation.tailrec
  * example between an asynchronous producer and consumer.
  */
 final class FiberRefs private (
-  private[zio] val fiberRefLocals: Map[FiberRef[_], (::[(FiberId.Runtime, Any, Int)], Int)]
+  private[zio] val fiberRefLocals: Map[FiberRef[_], FiberRefs.Value]
```

Comment:
I feel like using a Scala immutable map is ultimately the wrong choice here, but that's a separate ticket. 😆

Reply:
Definitely agree!

Reply:
What would you use instead? 🤔

Reply:
@guizmaii  I'll need to think this through, but I think we can splice in some sort mutability and copy things over whenever we fork. Having said that, mutable.HashMap doesn't seem to be the answer either because copying it is too expensive when we fork. I'm actually wondering whether we even need a Map here since the number of entries in this Map is most of the time very small

Reply:
In fact, we're not even really using a HashMap in the benchmarks since the number of `FiberRef`s is `<=4`, so we're using Scala's `MapX` implementations which just iterate through the contents of the collection